### DISCUSSION TITLE
Ability to hide "view in CloudWatch console” link from result

### DISCUSSION THREAD

**Discussion by ainapudi** [grafana/grafana#44940] [Data Sources]
Our users will not have access to AWS console. We want the ability to hide "view in CloudWatch console” link from result. Instead the link should jump to a native Grafana interface to look for log lines around this line.

**Comment by mchene** [comment]
Thank you for the feedback. The “view in CloudWatch console” link feature is meant to deep link to the AWS console in context (e.g., time) with the query used to generate the panel data to further investigate the data (metrics or logs) natively in the AWS console. The ability to allow a dashboard editor to overwrite this link in a panel resulting from a CloudWatch query is not something we are currently planning but something we may be able to consider in the future.
In regards to “Instead the link should jump to a native Grafana interface to look for log lines around this line.”, dashboard editors can add their own set of links to any panel that I believe can solve this request, see “[Add a link to a panel](https://grafana.com/docs/grafana/latest/panels/working-with-panels/add-link-to-panel/)” documentation.

**Comment by ainapudi** [comment]
Editor cannot change the link from the CW search result list. Grafana users will not always have access to AWS console.

**Comment by taleena** [comment]
Unless I'm mistaken, I think Marc's suggestion was to add a brand new link using the documentation he provided, and not to change the existing CW link. This, overall, sounds like a separate issue from hiding the CW link - viewing log lines associated with a CW metric within Grafana. If you'd like to explore more about this one, please file a new discussion, and we'd want to know what you're currently seeing and what you expect.

It sounds like there could be a couple of different solutions for removing the CW console link request, and we'd need to know more about to make a better assessment. @ainapudi do you have SOME users that have access to the AWS console that would find this link to CW beneficial? Or, is your request that NONE of your users are able to use that link, or even see it? 

Would you be able to provide the screenshots of the link so we can be crystal clear on the locations, both the "CW search result list" and anywhere else you are referencing?

**Comment by ainapudi** [comment]
NONE of your users should see that link. The "View in CloudWatch console" links in the attached image.
![image (1)](https://user-images.githubusercontent.com/97984672/166077544-afd83357-939d-4788-b3e8-3590a3147e7d.png)

**Comment by taleena** [comment]
Thanks for the added context @ainapudi, to make sure I understand, this is not using an Amazon-managed Grafana instance, right? @mchene I think the comment above is at odds with what you said, since it sounds like the request is to blanket remove it, but perhaps I'm misunderstanding the context/use cases.

**Comment by ainapudi** [comment]
Right, this is not using an Amazon-managed Grafana instance.

**Comment by n-benchoubane** [comment]
Any updates on this? I wish to take off the links to cloudwatch as well.

**Comment by notbrain** [comment]
Also looking for an option to turn off all "View in CloudWatch" links in tables generated by CW Insights queries etc.

**Comment by matschaffer-roblox** [comment]
Looks like you can add a regex override rule with an empty data links set to disable the default linking behavior.

![Screen_Shot_2023-03-29_at_4_23_26_PM](https://user-images.githubusercontent.com/115565899/228689698-6552b84e-467e-4be7-b7bb-e624736b66f0.png)

**Reply by Vaibhav-1995** [reply]
@matschaffer-roblox 

Thank you!
This helped me out and works with the cloudwatch metrics Timeseries and Stats panels.

But for Cloudwatch Logs dashboard in logs panel we don't have option to add Overrides Field, so not able to remove "View in Cloudwatch Console" link from logs as below - 
![Screenshot (55)](https://github.com/grafana/grafana/assets/103194544/7970575d-d33d-45c6-ab6b-706b6dfa78b1)

Could you please help to remove this link from Logs Dashboard as well.

**Comment by taleena** [comment]
@Vaibhav-1995 Ha! I was just responding to ask you to open an issue, but went searching and found you did already! Thank you for doing that. I'm linking mentioning it here: https://github.com/grafana/grafana/issues/79910

I did notice you are on a quite older version, so a quick note that adding this would require you to update to whatever future version it ends up on. It might be easier for you to get some of that effort out of the way now since we're already in v10.

**Comment by katebrenner** [comment]
closing in favor of github issues - see https://github.com/grafana/grafana/issues/79910
